lib/manifestTools/validationRules/all/apiAccessRulesValid.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var validationConstants = require('../../validationConstants');
+var validations = require('../../../common/validations');
+var validAccessTypes = {
+  'ios': [ 'cordova' ],
+  'windows': [ 'cordova' ],
+  'android': [ 'cordova' ],
+  'windows10': [ 'all', 'allowForWebOnly' ]
+};
+
+// Add 'none' as a valid API access type for all platforms
+for (var platform in validAccessTypes) {
+  if (validAccessTypes[platform].indexOf('none') < 0) {
+    validAccessTypes[platform].push('none');
+  }
+}
+
+module.exports = function (manifestContent, callback) {
+  var validationResults = [];
+  var apiAccessRules = manifestContent.mjs_api_access;
+
+  if (apiAccessRules && apiAccessRules instanceof Array) {
+  	apiAccessRules.forEach(function (rule, index) {    
+      if (rule.platform) {
+        rule.platform.split(';')
+            .map(function (item) { return item.trim(); })
+            .forEach(function (platform) {

```

Comment:
Consider using .reduce() instead of forEach() here.  Essentially:
```JS
var  validationResults = apiAccessRules
  .map( ... trim... )
  .filter( ... valid platforms... )
  .reduce( function(results, nextItem ){ .... make validation object... ; return result; }, []);
```